pkgdb/src/resolver/environment.cc

```diff
@@ -635,23 +688,11 @@ Environment::lockSystem( const System & system )
                 throw ResolutionFailureException(
                   "no inputs found to search for packages" );
               }
-            /* Print group name if this isn't the default group. */
-            if ( auto descriptor = group->begin();
-                 descriptor != group->end()
-                 && descriptor->second.group.has_value() )
-              {
-
-                msg << "  in group `" << *descriptor->second.group << "': ";
-              }
-            else { msg << "  in default group:"; }
-            msg << std::endl;
-            for ( const auto & [iid, url] : failure )
-              {
-                msg << "    failed to resolve `" << iid << "' in input `" << url
-                    << '\'';
-              }
 
-            ++group;
+            /* Describe the failure. */
+            describeResolutionFailure( groupMsg, *group, failure );
+            throw ResolutionFailureException(
```

Comment:
suggestion blocking: I don't think we need to throw here anymore without the try, and groupMsg isn't really used. Should this be `describeResolutionFailure( msg, *group, failure );`?

Reply:
Cool I can make that change soon, thanks for the tip

Reply:
It looks like we do require that `throw`.
I get an infinite loop without it.

Reply:
Yeah that's because you dropped `++group`

You originally wrote the loop to print every resolution failure - this PR is changing it to print the first failure. I don't have strong feelings and am fine with either.

If you're going to leave as is and print all errors, we can drop the `throw` and put the `++group` back in.

If you're going to print the failure eagerly, we need to drop `if ( ! groups.empty() ) { throw ResolutionFailureException( msg.str() ); }` because it's dead code, and then `msg` is unused and should be dropped as well. And nonblocking, but I'd drop `group = groups.erase( group );` because it just makes the loop hard to read.

Reply:
Aight I think we're good now.